classes/chat/ChatManager.php

```diff
@@ -142,6 +150,45 @@ private function fetchPosts(?int $starting_post_id = null, int $max_posts = self
             }
             $post->message = nl2br($this->system->html_parse($post->message, false, true));
 
+            // Handle Quotes
+            $pattern = "/\[quote:\d+\]/";
+            $has_quote = preg_match_all($pattern, $post->message, $matches);
+            if ($has_quote) {
+                if ($allow_quote) {
+                    foreach ($matches[0] as $match) {
+                        // get chat post contents
+                        $pattern = "/\[quote:(\d+)\]/";
+                        preg_match($pattern, $match, $id);
+                        $quote = $this->fetchPosts($id[1], 1, false);
+                        if ($quote) {
+                            // if id match
+                            if ($quote[0]->id == $id[1]) {
+                                // format each entry in $quotes
+                                $formatted_quote = "<div class='quote_container'><a class='chat_user_name " . implode(' ', $quote[0]->user_link_class_names) . "' href='" . $this->system->router->getURL("members", ["user" => $quote[0]->user_name]) . "'>" . $quote[0]->user_name . "</a><div class='quote_message'>" . $quote[0]->message . "</div></div>";
+                                // run replace on $test using $matches as search and $quotes as replace
```

Comment:
This is okay for now but I think we should move more in the direction of embedding display logic in the front end, detecting in the UI component when there's a `quote` tag and replacing it with our desired markup. 

We should also probably drop the current BBcode system and migrate to a completely front-end system using something like Markdown https://github.com/remarkjs/react-markdown, so we can take care of it then.

Reply:
Marking as to-do.